dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -555,19 +587,28 @@ class _RecordingWidgetsBinding extends BindingBase
 
   @override
   void handleBeginFrame(Duration rawTimeStamp) {
+    // Don't keep on truckin' if there's an error.
+    if (_hasErrored) {
+      return;
+    }
```

Comment:
Instead of checking for `_hasErrored` in multiple places, what if we check for it in `_shouldContinue()` and rely on existing logic to stop pumping frames?

Reply:
I can't think of a clean way to use it. Currently `_shouldContinue` depends on everything being healthy. For example, it extracts the profile from rendered frames. It only communicates two signals: "continue running" and "benchmark finished successfully". We'd have to add a third value communicating "benchmark errored and must halt". When an error happens the system could be in a corrupted state, in which case our job is to halt immediately and report the error before it's obscured by something else. IOW I think they have sufficiently different roles to warrant separate codepaths and separate signals.